src/NUnitFramework/framework/Api/FrameworkController.cs

```diff
@@ -105,8 +98,8 @@ public FrameworkController(Assembly assembly, string idPrefix, IDictionary setti
         /// <param name="builderType">The Type of the test builder</param>
         public FrameworkController(string assemblyNameOrPath, string idPrefix, IDictionary settings, string runnerType, string builderType)
         {
-            Builder = (ITestAssemblyBuilder)Activator.CreateInstance(Type.GetType(builderType));
-            Runner = (ITestAssemblyRunner)Activator.CreateInstance(Type.GetType(runnerType), new object[] { Builder });
+            Builder = (ITestAssemblyBuilder)Reflect.Construct(Type.GetType(builderType));
```

Comment:
Not now because it would be a larger change, but it would be nice to switch Reflect.Construct to a generic method. Instead of `public static object Construct(Type type)` make it `public static T Construct<T>()`.